File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode487.java
import java.util.Scanner;

public class Leetcode487 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(findMaxConsecutiveOnes(arr));
    }

    private static int findMaxConsecutiveOnes(int[] nums) {
        int left = 0, sum = 0, ans = 0;
        for (int right = 0; right < nums.length; right++) {
            sum += nums[right];
            while (!(sum == right - left + 1 || sum == right - left)) {
                sum -= nums[left];
                left++;
            }
            ans = Math.max(ans, right - left + 1);
        }
        return ans;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode487.ts
function findMaxConsecutiveOnes(nums: number[]): number {
    let left: number = 0, sum: number = 0, ans: number = 0;
    for (let right: number = 0; right < nums.length; right++) {
        sum += nums[right];
        while (!(sum == right - left + 1 || sum == right - left)) {
            sum -= nums[left];
            left++;
        }
        ans = Math.max(ans, right - left + 1);
    }
    return ans;
}

let arr: number[] = [1, 1, 0, 1, 0, 1];
console.log(findMaxConsecutiveOnes(arr));

export { };